### PR TITLE
Fix a couple of NREs in completion controller when interacting with REPL commands in the Interactive Window

### PR DESCRIPTION
**Pull request by DustinCampbell**

### CODE REVIEW

**Comment by DustinCampbell** [comment]
Fixes issue #3962.

Reviewers: @amcasey, @KevinH-MS, @rchande, @dpoeschl, @Pilchie

**Comment by DustinCampbell** [comment]
jenkins, retest this please

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Backspace.cs [hunk 1]
```diff
@@ -90,9 +90,11 @@ private void ExecuteBackspaceOrDelete(ITextView textView, Action nextHandler, bo
 
                 var model = sessionOpt.Computation.InitialUnfilteredModel;
 
-                if ((model == null && CaretHasLeftDefaultTrackingSpan(subjectBufferCaretPoint, documentBeforeDeletion)) ||
+                var completionService = CreateCompletionService();
```
Comment (amcasey) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Backspace.cs]: Why pull out a local?  Doesn't that make it less lazy?

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs [hunk 2]
```diff
@@ -121,22 +121,29 @@ private void Commit(CompletionItem item, TextChange textChange, Model model, cha
...
+                        var changes = formattingService.GetFormattingChangesAsync(document, textChange.Span, CancellationToken.None).WaitAndGetResult(CancellationToken.None);
+                        document.Project.Solution.Workspace.ApplyTextChanges(document.Id, changes, CancellationToken.None);
+                        formattingTransaction.Complete();
+                    }
                 }
             }
 
             // Let the completion rules know that this item was committed.
-            GetCompletionRules().CompletionItemCommitted(item);
+            var completionRules = GetCompletionRules();
```
Comment (amcasey) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs]: Use ```?.```?

**Comment by amcasey** [comment]
Regression test?

**Comment by amcasey** [comment]
LGTM

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs (hunk 2 above)
Comment (DustinCampbell) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs]: Yes! Sorry. Need to get that into my C# muscle memory a bit better.

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Backspace.cs (hunk 1 above)
Comment (DustinCampbell) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Backspace.cs]: It does break short-circuiting. Will fix.

**Comment by rchande** [comment]
Agree with @amcasey that it would be good to have a test around this.

Otherwise LGTM.

**Comment by DustinCampbell** [comment]
Looking at adding a regression test.

**Comment by DustinCampbell** [comment]
FWIW, I think I'm going to abandon this change. Really, it just band-aids the underlying problem of there not being a document in the workspace for "Interactive Command", despite it seemingly being registered correctly as a language service. I'm going to dig in there and see if I can solve the root cause.

**Comment by DustinCampbell** [comment]
Abandoning in favor of PR #4088

**Pull request closed without merging**